recipes/sbp/all/test_package/CMakeLists.txt

```diff
@@ -0,0 +1,8 @@
+cmake_minimum_required(VERSION 3.1)
+project(test_package C)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+add_executable(${PROJECT_NAME} test_package.c)
```

Comment:
I guess I should instead be doing `add_subdirectory` for https://github.com/swift-nav/libsbp/tree/master/c/examples

Reply:
The samples depend on eigen3 (which is eigen here) and https://github.com/swift-nav/libsbp/tree/master/c/examples/example requires libserialport, which I've created https://github.com/conan-io/conan-center-index/issues/6703 for.

We'll need to enhance the cmake files for the examples to be able to get them building without libserialport, and also enhance the example code so some of them are usable with static data (i.e. not data from tcp or serial port)

Reply:
A test package must be self contained. It should not depend on a directory in the libbsp's sources folder.
Having a very simple source is fine.

Something like:

```c
#include <libsbp/sbp.h>
#include <stdlib.h>

int main(int argc, char *argv[]) {
    struct sp_port port = NULL;
    int result;
    if (argc < 2) {
        fprintf(stderr, "Need an argument: %s SERIALPORT\n", argv[0]);
        return EXIT_FAILURE;
    }

    result = sp_get_port_by_name(argv[1], &port);
    if (result != SP_OK) {
        fprintf(stderr, "sp_get_port_by_name failed (ignored)\n");
        return EXIT_SUCCESS; // Ignore these kind of errors in a test package
    }
    printf("sp_get_port_by_name succeeded\n");

    result = sp_open(port, SP_MODE_READ);
    if (result != SP_OK) {
        fprintf(stderr, "sp_open failed (ignored)\n");
        return EXIT_SUCCESS; // Ignore these kind of errors in a test package
    }
    printf("sp_open succeeded\n");

    result = sp_close(port);
    if (result != SP_OK) {
        fprintf(stderr, "sp_close failed (ignored)\n");
        return EXIT_SUCCESS; // Ignore these kind of errors in a test package
    }
    printf("sp_close succeeded\n");

    sp_free_port(port);
    return EXIT_SUCCESS;
}
```
I don't think there's a need for `libserialport.h` in the test package.

Reply:
I'm confused.  Your example code depends on `libserialport`.

I can make the `test_package.c` use SBP functions without needing `libserialport`, or a TCP server, but that requires a SBP raw file as input.  It can be very small, e.g. https://github.com/swift-nav/libsbp/blob/master/test_data/short.sbp , but smaller is possible.  Can the test data come from the source repo, or should it be in the conan repo?

Reply:
Oh sorry! I thought the `sp_` functions came from libbsp.

Reply:
It looks like libsbp provides executables to create these binary SBP files.
You can do the following:
- provide a `data.json`

```python
    def test(self):
        if not tools.cross_building(self):
            data_bsp = os.path.join(self.source_folder, "data.bsp")
            data_bsp = os.path.join(self.build_folder, "data.bsp")
            self.run("json2bsp {} -o {}".format(os.path.join(self.source_folder, "data.json"), data_bsp)  # Wrong arguments
            self.run("{} {}".format(os.path.join("bin", "test_package"), data_bsp), run_environment=True)
```

Reply:
why `if not tools.cross_building(self)` ? The data format is platform neutral.

Reply:
The `json2bsp` executable can only execute when not cross building.